kernel/comps/virtio/src/transport/mod.rs

```diff
@@ -94,6 +104,115 @@ pub trait VirtioTransport: Sync + Send + Debug {
     ) -> Result<(), VirtioTransportError>;
 }
 
+/// Manage PCI device/notify configuration space (legacy/modern).
+#[derive(Debug)]
+pub struct ConfigManager<T: Pod> {
+    safe_ptr: Option<SafePtr<T, IoMem>>,
```

Comment:
The `safe_ptr` may not be a good choice. Since we use this field to determine if the device is modern, I suggest renaming it to `modern_config_ptr` or another name that includes `modern`.